sycl/source/detail/event_impl.cpp

```diff
@@ -363,6 +370,116 @@ uint64_t event_impl::get_profiling_info<info::event_profiling::command_end>() {
+    std::shared_ptr<ext::oneapi::experimental::detail::node_impl> NodeImpl) {
+  if (!isEventFromSubmittedExecCommandBuffer()) {
+    throw sycl::exception(sycl::make_error_code(sycl::errc::invalid),
+                          "Node profiling information is only available for "
+                          "event returned from graph submission" +
+                              codeToString(PI_ERROR_INVALID_VALUE));
+  }
+  checkProfilingPreconditions();
+  // Node profiling is only available for Level-Zero backend.
+  if (MContext->getBackend() != backend::ext_oneapi_level_zero) {
+    throw sycl::exception(
+        sycl::make_error_code(sycl::errc::invalid),
+        "Node Profiling info is only available for Level-Zero backend. " +
+            codeToString(PI_ERROR_PROFILING_INFO_NOT_AVAILABLE));
+  }
+  if (!MHostEvent) {
+    if (MEvent) {
+      assert((MExecGraph) && "Get Node Profiling Info: MExecGraph is empty");
+      sycl::detail::pi::PiExtSyncPoint SyncPoint =
+          MExecGraph->getSyncPointFromNode(NodeImpl);
+
+      auto StartTime =
+          get_event_profiling_info<info::event_profiling::command_start>(
+              this->getHandleRef(), SyncPoint, this->getPlugin());
+      if (!MFallbackProfiling) {
+        return StartTime;
+      } else {
+        auto DeviceBaseTime =
+            get_event_profiling_info<info::event_profiling::command_submit>(
+                this->getHandleRef(), this->getPlugin());
+        return MHostBaseTime - DeviceBaseTime + StartTime;
+      }
+    }
+    return 0;
+  }
+  if (!MHostProfilingInfo)
+    throw sycl::exception(
+        sycl::make_error_code(sycl::errc::invalid),
+        "Profiling info is not available. " +
+            codeToString(PI_ERROR_PROFILING_INFO_NOT_AVAILABLE));
+  return MHostProfilingInfo->getStartTime();
+}
+
+template <>
+uint64_t event_impl::get_profiling_info<info::event_profiling::command_end>(
+    std::shared_ptr<ext::oneapi::experimental::detail::node_impl> NodeImpl) {
+  if (!isEventFromSubmittedExecCommandBuffer()) {
+    throw sycl::exception(sycl::make_error_code(sycl::errc::invalid),
+                          "Node profiling information is only available for "
+                          "event returned from graph submission" +
+                              codeToString(PI_ERROR_INVALID_VALUE));
+  }
+  checkProfilingPreconditions();
+  // Node profiling is only available for Level-Zero backend.
+  if (MContext->getBackend() != backend::ext_oneapi_level_zero) {
+    throw sycl::exception(
+        sycl::make_error_code(sycl::errc::invalid),
+        "Node Profiling info is only available for Level-Zero backend. " +
+            codeToString(PI_ERROR_PROFILING_INFO_NOT_AVAILABLE));
+  }
+  if (!MHostEvent) {
+    if (MEvent) {
+      assert((MExecGraph) && "Get Node Profiling Info: MExecGraph is empty");
+      sycl::detail::pi::PiExtSyncPoint SyncPoint =
+          MExecGraph->getSyncPointFromNode(NodeImpl);
```

Comment:
should we check the return value here  isn't `-1`? i.e No sync-point found for node.

Reply:
Actually this code didn't work because `sycl::detail::pi::PiExtSyncPoint` is defined as an `unsigned int`. I have adapted the code to throw an exception if the node is not found.